### PR TITLE
add combobox to explicitly set mail encryption mode

### PR DESCRIPTION
**Pull request by lfuelling**
Hi,

this PR replaces the "Use SSL/TLS" switch in the mailserver config with a radio group that enables to explicitly set the SSL mode to use. 

The following radio choices are available:
- No encryption
    - The previous behavior when the switch was turned off
    - This is the default selection (like it was before)
- SSL/TLS
- STARTTLS
- Auto
    - The previous behavior when the switch was turned on

The PR is `WIP` because of the following leftover questions:
- [x] Is a migration really required for this?
    - [original comment](https://github.com/openmediavault/openmediavault/issues/536#issuecomment-567043344)
- [x] ~~The Contribution guidelines say I have to "subscribe the CLA online via GitHub"~~
    - I think "sign" is meant here
    - ~~I don't know how to sign the CLA on GitHub~~

### CODE REVIEW

**Comment by votdev** [comment]
Please add the following line to your commit: `Fixes: https://github.com/openmediavault/openmediavault/issues/536`

**Comment by lfuelling** [comment]
It's already in the commit. Above the `Signed-off-by` line.

**Comment by votdev** [comment]
Oh right, i am usually used to see this in the PR description, too.

**Comment by votdev** [comment]
You need to adapt the database and RPC datamodel:
* https://github.com/openmediavault/openmediavault/blob/master/deb/openmediavault/usr/share/openmediavault/datamodels/conf.system.notification.email.json#L23
* https://github.com/openmediavault/openmediavault/blob/master/deb/openmediavault/usr/share/openmediavault/datamodels/rpc.emailnotification.json#L21

**Comment by votdev** [comment]
>     * Is a migration really required for this?

Yes, the property type has been changed from boolean to a string enum.

**Comment by votdev** [comment]
Please use a `combo` box instead a radio group (this is not used in the whole UI).

File: deb/openmediavault/usr/share/openmediavault/datamodels/conf.system.notification.email.json [hunk 1]
```diff
@@ -21,8 +21,8 @@
 			"default": 25
 		},
 		"tls": {
-		    "type": "boolean",
-			"default": false
+		    "type": "string",
```
Comment (votdev) [inline deb/openmediavault/usr/share/openmediavault/datamodels/conf.system.notification.email.json]: Please use a string enum which limits the possible values to the minimum.
Example:
```
"execution":{
			"type": "string",
			"enum": [
				"exactly",
				"hourly",
				"daily",
				"weekly",
				"monthly",
				"yearly",
				"reboot"
			],
			"default": "exactly"
		},
```

File: deb/openmediavault/srv/salt/omv/deploy/postfix/files/main.cf.j2 [hunk 2]
```diff
@@ -37,8 +37,8 @@ smtp_sasl_security_options = {{ smtp_sasl_security_options }}
 smtp_sasl_password_maps = hash:{{ smtp_sasl_password_maps }}
 smtp_sasl_mechanism_filter = {{ smtp_sasl_mechanism_filter }}
 {%- endif %}
-{%- if config.tls | to_bool %}
-{%- if config.port == 465 %}
+{%- if config.tls == 'ssl' or config.tls == 'starttls' or config.tls == 'auto' %}
+{%- if config.tls == 'starttls' or (config.tls == 'auto' and config.port == 465) %}
```
Comment (lfuelling) [inline deb/openmediavault/srv/salt/omv/deploy/postfix/files/main.cf.j2]: This line is three chars longer than recommended, I hope that's okay.

Comment (votdev) [reply deb/openmediavault/srv/salt/omv/deploy/postfix/files/main.cf.j2]: Yes, [no problem](https://github.com/openmediavault/openmediavault/blob/master/.editorconfig#L24) (doing this in Jinja files will result in expected unexpect behaviour :-)).

File: deb/openmediavault/usr/share/openmediavault/datamodels/rpc.emailnotification.json [hunk 3]
```diff
@@ -19,7 +19,7 @@
 				"required": true
 			},
 			"tls": {
-			    "type": "boolean",
+			    "type": "string",
```
Comment (votdev) [inline deb/openmediavault/usr/share/openmediavault/datamodels/rpc.emailnotification.json]: Please define the enum here, too.
```
"enum": [
				"no",
				"ssl",
				"starttls",
				"auto"
			],
```

**Comment by lfuelling** [comment]
The force push was necessary to add the required lines to the merge commit mesage.

**Comment by votdev** [comment]
Oh please squash to only 1 commit.

**Comment by votdev** [comment]
Now only the migration script is missing, then everything should be ok.

**Comment by lfuelling** [comment]
Regarding the migration script:
What method should I use for it? 
I can't find only `omv_config_update` and that seems to update a value, not it's type.

File: deb/openmediavault/var/www/openmediavault/js/omv/module/admin/system/notification/Notification.js [hunk 4]
```diff
@@ -175,10 +175,19 @@ Ext.define("OMV.module.admin.system.notification.Settings", {
 					text: _("The default SMTP mail server port, e.g. 25 or 587.")
 				}]
 			},{
-				xtype: "checkbox",
-				name: "tls",
-				fieldLabel: _("Use SSL/TLS secure connection"),
-				checked: false
+				xtype: 'combobox',
```
Comment (votdev) [inline deb/openmediavault/var/www/openmediavault/js/omv/module/admin/system/notification/Notification.js]:
```suggestion
				xtype: "combo",
				fieldLabel: _("Encryption mode"),
				queryMode: "local",
				displayField: "name",
				valueField: "value",
				editable: false,
				name: "tls",
				store: [
					[ "no", _("No encryption") ],
					[ "ssl", _("SSL/TLS") ],
					[ "starttls", _("STARTTLS") ],
					[ "auto", _("Auto") ]
				]
```

Comment (lfuelling) [reply deb/openmediavault/var/www/openmediavault/js/omv/module/admin/system/notification/Notification.js]: The [docs](https://docs.sencha.com/extjs/7.0.0/classic/Ext.form.field.ComboBox.html) read like both is possible but I'll change it.

**Comment by votdev** [comment]
Closed in favor of https://github.com/openmediavault/openmediavault/pull/540.

**Pull request closed without merging**